src/CommitmentsV2/SFA.DAS.CommitmentsV2.Api/DependencyResolution/DefaultRegistry.cs

```diff
@@ -14,6 +15,7 @@ public DefaultRegistry()
             For<IDbContextFactory>().Use<SynchronizedDbContextFactory>();
             For<IAuthenticationService>().Use<AuthenticationService>().Singleton();
             For<IModelMapper>().Use<ModelMapper>();
+            For<IAlertsMapper>().Use<AlertsMapper>();
```

Comment:
@chrisfoster76 im not sure if this is in the correct place, can you give a pointer as to where this might need to live? I think it has been done in this way as what it is returning isn't standard, and it might be due to it being a mapper in a mapper 💥

Reply:
What about if those private datalock methods eg, `HasCourseDataLockPendingChanges` were extension methods for `DataLockStatus`. Then a `GetAlerts` extension method against the `Apprenticeship` model could wrap up some of the other bits. `AlertsMapper` can then be removed. A new `[Flags]` enum as the return type will also help ;-)

Reply:
v1 had some reasonable extension methods in this sort of area

Reply:
Also this approach will allow us to get the benefit as we will need the same logic

Reply:
Have removed this - and changed as your suggestion